### PR TITLE
finished lesson 6

### PR DESCRIPTION
**Pull request by yixingxu**
used pytest to do the testing

### CODE REVIEW

**Comment by natasha-aleksandrova** [comment]
@yixingxu please take a look at the files in this PR.

before running `git add .` it is always a good idea to run `git status`

I cannot merge this PR with the files outside of your student directory as this may cause issues for other students.

You may try `git revert` or `git rm`

File: students/yixingxu/lesson06/test_mailroom4.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    assert (expected == mailroom4.create_a_report()) is True
+    
+def test_create_summary():
+    donation_history = [
+                    {'name':'Anna','donations':[100,200,300]},
+                    {'name':'Bob','donations':[1000,2000]},
+                    {'name':'Chuck','donations':[10000]},
+                    {'name':'David','donations':[1]},
+                    {'name':'Ethan','donations':[10,20]}  
+                    ]
```
Comment (natasha-aleksandrova) [inline students/yixingxu/lesson06/test_mailroom4.py]: this isn't used.

File: students/yixingxu/lesson06/mailroom4.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    return list[2]
+    
+# quit
+def quit_program():
+    print('Quit')
+    return 'exit menu'
+
+def send_letters_to_everyone(donation_history = donation_history):
+    for item in donation_history:     
+        Letter = "Dear {name},\n\n        Thank you for all your very kind donations of ${donations}\n\n        It will be put to very good use.\n\n                       Sincerely,\n                          -The Team".format(**item)
```
Comment (natasha-aleksandrova) [inline students/yixingxu/lesson06/mailroom4.py]: make sure to use lower case for var names.

File: students/yixingxu/lesson06/mailroom4.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+# get the total given amount 
+def sort_by_total_given(list):
+    return list[2]
+    
+# quit
+def quit_program():
+    print('Quit')
+    return 'exit menu'
+
+def send_letters_to_everyone(donation_history = donation_history):
```
Comment (natasha-aleksandrova) [inline students/yixingxu/lesson06/mailroom4.py]: this would be a good function to test

File: students/yixingxu/lesson06/mailroom4.py (hunk 2 above)
Comment (yixingxu) [reply students/yixingxu/lesson06/mailroom4.py]: missed one. Thanks. fixed this one.

File: students/yixingxu/lesson06/mailroom4.py (hunk 3 above)
Comment (yixingxu) [reply students/yixingxu/lesson06/mailroom4.py]: can you give me a hint. How to test written out files? Thanks!

Comment (natasha-aleksandrova) [reply students/yixingxu/lesson06/mailroom4.py]: in your test call `send_letters_to_everyone()`
then assert that files exists via `os.path.isfile`, in addition you can also open the file and assert contents of the file

File: students/yixingxu/lesson06/test_mailroom4.py (hunk 1 above)
Comment (yixingxu) [reply students/yixingxu/lesson06/test_mailroom4.py]: function create_summary() has a default input, which i why I put the default input donation_history in the test function. create_summary() is from line 99 to line 106 in mailroom4.py

File: students/yixingxu/lesson06/mailroom4.py (hunk 3 above)
Comment (yixingxu) [reply students/yixingxu/lesson06/mailroom4.py]: Thanks!

File: students/yixingxu/lesson06/test_mailroom4.py (hunk 1 above)
Comment (natasha-aleksandrova) [reply students/yixingxu/lesson06/test_mailroom4.py]: screen shot 2018-05-22 at 3 31 35 pm

**Pull request merged**